apps/aecore/lib/aecore/naming/tx/name_pre_claim_tx.ex

```diff
@@ -45,12 +45,12 @@ defmodule Aecore.Naming.Tx.NamePreClaimTx do
 
   # Callbacks
 
-  @spec init(payload() | map()) :: NamePreClaimTx.t()
-  def init(%{commitment: %Identifier{} = identified_commitment} = _payload) do
+  @spec init(payload()) :: NamePreClaimTx.t()
```

Comment:
@spec init(payload() | map()) :: NamePreClaimTx.t()

Reply:
why so, isn't payload a map?

Reply:
the payload is: `%{commitment: binary{}}` or you can change the `@type payload` to:   
``` elixir
@type payload :: %{
          commitment: commitment_hash() | Identifier.t()
        }
```

Reply:
I still don't get it, why both payload() and map()

Reply:
Is better to change the payload to:
``` elixir
@type payload :: %{
          commitment: commitment_hash() | Identifier.t()
        }
```
and leave the spec as it is:
``` elixir
 @spec init(payload()) :: NamePreClaimTx.t()
```